### PR TITLE
update after 0.3.0 release

### PR DESCRIPTION
**Pull request by loosebazooka**
Signed-off-by: Appu Goundan <appu@google.com>

### CODE REVIEW

**Review by vlsi — APPROVED** [review]

File: sigstore-gradle/sigstore-gradle-sign-base-plugin/src/main/kotlin/dev/sigstore/sign/SigstoreSignExtension.kt [hunk 1]
```diff
@@ -39,7 +39,7 @@ abstract class SigstoreSignExtension(private val project: Project) {
     abstract val sigstoreJavaVersion : Property<String>
 
     init {
-        sigstoreJavaVersion.convention("0.2.0")
+        sigstoreJavaVersion.convention("0.3.0")
```
Comment (vlsi) [inline sigstore-gradle/sigstore-gradle-sign-base-plugin/src/main/kotlin/dev/sigstore/sign/SigstoreSignExtension.kt:42]: Looks like this should be "before 0.3.0 release"

**Review by vlsi — DISMISSED** [review]

File: sigstore-gradle/sigstore-gradle-sign-base-plugin/src/main/kotlin/dev/sigstore/sign/SigstoreSignExtension.kt (hunk 1 above)
Comment (loosebazooka) [reply sigstore-gradle/sigstore-gradle-sign-base-plugin/src/main/kotlin/dev/sigstore/sign/SigstoreSignExtension.kt:42]: I'm not sure I understand. Also, I'm not sure I understand these test error messages.

Comment (vlsi) [reply sigstore-gradle/sigstore-gradle-sign-base-plugin/src/main/kotlin/dev/sigstore/sign/SigstoreSignExtension.kt:42]: I think the change of return type from String to byte array was not really backward compatible: https://github.com/sigstore/sigstore-java/commit/f3b363d2f0c2c947c894c3ecbafbfc0f3ab4a463#diff-13a2e26b591d0577e3c7a3f31beedf30cf2645b61d4179ce14407d7157279ff6

Comment (loosebazooka) [reply sigstore-gradle/sigstore-gradle-sign-base-plugin/src/main/kotlin/dev/sigstore/sign/SigstoreSignExtension.kt:42]: hah yeah I get that. But I don't think I understand why CI is failing but my local isn't.

Comment (loosebazooka) [reply sigstore-gradle/sigstore-gradle-sign-base-plugin/src/main/kotlin/dev/sigstore/sign/SigstoreSignExtension.kt:42]: oh yeah, there's a separate config for CI.

Comment (vlsi) [reply sigstore-gradle/sigstore-gradle-sign-base-plugin/src/main/kotlin/dev/sigstore/sign/SigstoreSignExtension.kt:42]: Do you mean `CI=Y` environment var?
Should the test print something like "Running reduced set of tests to get faster feedback, consider adding CI=Y to execute all the tests" ?

**Review by vlsi — APPROVED** [review]

**Comment by loosebazooka** [comment]
@vlsi I think I don't understand the version magic here with the gradle plugin. Obvioulsy our API is not backwards compatible here across changes, but what's triggering the issue?

**Comment by vlsi** [comment]
I will check tomorrow

**Comment by vlsi** [comment]
It works as planned, except, it was not prepared that you would continue development after releasing the version.

Here's how it works.
It tests Gradle plugin with several versions of sigstore-java:
1) Currently built sigstore-java
2) The default sigstore-java (e.g. if you apply Gradle plugin without explicitly mentioning sigsore-java version)
3) The minimal "supported" sigstore-java. For now, it is explicitly mentioned in https://github.com/sigstore/sigstore-java/blob/99ceb8388cb7354bfef8db8a185c55bd81949230/sigstore-testkit/src/main/kotlin/dev/sigstore/testkit/BaseGradleTest.kt#L70

`1)` is always tested (locally and on CI)
`2)` and `3)` are tested on CI only to save time (and OIDC flows) when testing locally.

Now there's a question: how does it distinguish between "currently built sigstore-java" and "sigstore-java from Central"?
Well, for now it does so by comparing the requested `sigstore-java` version and the version mentioned in `gradle.properties`.
If the requested version matches the one mentioned in `gradle.properties`, the test assumes it needs to test the locally built version.
As you might guess, the version in `gradle.properties` have not been updated after release, so the PRs were all green even after introducing backward-incompatible changes.

----

I suggest making the following change, so it can properly distinguish between "locally built 0.3.0 and 0.3.0 from Central".
WDYT?

```diff
diff --git a/sigstore-testkit/src/main/kotlin/dev/sigstore/testkit/BaseGradleTest.kt b/sigstore-testkit/src/main/kotlin/dev/sigstore/testkit/BaseGradleTest.kt
index 5ff846f..7161c1b 100644
--- a/sigstore-testkit/src/main/kotlin/dev/sigstore/testkit/BaseGradleTest.kt
+++ b/sigstore-testkit/src/main/kotlin/dev/sigstore/testkit/BaseGradleTest.kt
@@ -42,7 +42,7 @@ open class BaseGradleTest {
         val EXTRA_LOCAL_REPOS  = System.getProperty("sigstore.test.local.maven.repo").split(File.pathSeparatorChar)

         val SIGSTORE_JAVA_CURRENT_VERSION =
-            TestedSigstoreJava.Version(
+            TestedSigstoreJava.LocallyBuiltVersion(
                 System.getProperty("sigstore.test.current.version")
             )

@@ -96,7 +96,10 @@ open class BaseGradleTest {
     protected fun declareRepositories(sigstoreJava: TestedSigstoreJava) =
         """
         repositories {${
-            if (sigstoreJava == SIGSTORE_JAVA_CURRENT_VERSION || sigstoreJava is TestedSigstoreJava.Default) {
+            if (sigstoreJava !is TestedSigstoreJava.LocallyBuiltVersion) {
+                ""
+            } else {
+                // Testing locally-built sigstore-java, so we add a repository for it
                 EXTRA_LOCAL_REPOS.joinToString("\n") {
                     """

@@ -107,8 +110,6 @@ open class BaseGradleTest {
                 }
                 """.trimIndent().prependIndent("            ")
                 }
-            } else {
-                ""
             }
         }
             mavenCentral()
@@ -123,6 +124,8 @@ open class BaseGradleTest {
                     TestedSigstoreJava.Default -> ""
                     is TestedSigstoreJava.Version ->
                         "sigstoreClient(\"dev.sigstore:sigstore-java:${sigstoreJava.version}\")"
+                    is TestedSigstoreJava.LocallyBuiltVersion ->
+                        "sigstoreClient(\"dev.sigstore:sigstore-java:${sigstoreJava.version}\")"
                 }
             }
         }
diff --git a/sigstore-testkit/src/main/kotlin/dev/sigstore/testkit/TestedSigstoreJava.kt b/sigstore-testkit/src/main/kotlin/dev/sigstore/testkit/TestedSigstoreJava.kt
index 32993d6..a66f016 100644
--- a/sigstore-testkit/src/main/kotlin/dev/sigstore/testkit/TestedSigstoreJava.kt
+++ b/sigstore-testkit/src/main/kotlin/dev/sigstore/testkit/TestedSigstoreJava.kt
@@ -28,9 +28,16 @@ sealed class TestedSigstoreJava {
     }

     /**
-     * Configures sigstore-java version for testing with Sigstore Gradle plugin.
+     * Configures taken from external repository sigstore-java version for testing with Sigstore Gradle plugin.
      */
     data class Version(
         val version: String,
     ): TestedSigstoreJava()
+
+    /**
+     * Configures locally-built sigstore-java version for testing with Sigstore Gradle plugin.
+     */
+    data class LocallyBuiltVersion(
+        val version: String,
+    ): TestedSigstoreJava()
 }
```

**Comment by loosebazooka** [comment]
superseded by #351

**Pull request closed without merging**